rollup/internal/controller/sender/sender.go

```diff
@@ -418,7 +492,7 @@ func (s *Sender) checkPendingTransaction() {
 		}
 
 		receipt, err := s.client.TransactionReceipt(s.ctx, tx.Hash())
-		if (err == nil) && (receipt != nil) { // tx confirmed.
+		if err == nil { // tx confirmed.
```

Comment:
What's the rationale for this change?

Reply:
To simplify the if-else logic here. The `TransactionReceipt` would not return this case (err == nil && receipt == nil).

now: https://github.com/scroll-tech/go-ethereum/blob/4553f5f2693542c20b5c68080ae5bacad12e4cf2/ethclient/ethclient.go#L288-L292
upstream: https://github.com/kumakichi/go-ethereum/blob/65142d81dd2723b54914233a21d35b3d6ac57832/ethclient/ethclient.go#L310-L312

if the code repo does not want to rely on this feature (maybe?) of `TransactionReceipt`, other places also need this `nil` check, in `TransactionByHash`, `BlockByNumber`, etc. So I think we can keep it simple (these APIs all return `ethereum.NotFound` instead of a nil pointer and a nil error).

Reply:
> To simplify the if-else logic here (IMO the current version is more readable). The `TransactionReceipt` would not return this case (err == nil && receipt == nil).
> 
> now: https://github.com/scroll-tech/go-ethereum/blob/4553f5f2693542c20b5c68080ae5bacad12e4cf2/ethclient/ethclient.go#L288-L292 upstream: https://github.com/kumakichi/go-ethereum/blob/65142d81dd2723b54914233a21d35b3d6ac57832/ethclient/ethclient.go#L310-L312
> 
> if the code repo does not want to rely on this feature (maybe?) of `TransactionReceipt`, other places also need this `nil` check, in `TransactionByHash`, `BlockByNumber`, etc. So I think we can keep it simple (these APIs all return `ethereum.NotFound` instead of a nil pointer and a nil error).